src/main/java/io/confluent/kafkarest/MetadataObserver.java

```diff
@@ -180,20 +181,24 @@ public int getLeaderId(final String topicName, final int partitionId) {
 
       Partition p = new Partition();
       p.setPartition(partId);
-      LeaderAndIsr leaderAndIsr =
-          zkUtils.getLeaderAndIsrForPartition(topic, partId).get();
-      p.setLeader(leaderAndIsr.leader());
-      scala.collection.immutable.Set<Integer> isr = leaderAndIsr.isr().toSet();
-      List<PartitionReplica> partReplicas = new Vector<PartitionReplica>();
-      for (Object brokerObj : JavaConversions.asJavaCollection(part.getValue())) {
-        int broker = (Integer) brokerObj;
-        PartitionReplica
-            r =
-            new PartitionReplica(broker, (leaderAndIsr.leader() == broker), isr.contains(broker));
-        partReplicas.add(r);
+      try {
+        LeaderAndIsr leaderAndIsr =
+                zkUtils.getLeaderAndIsrForPartition(topic, partId).get();
+        p.setLeader(leaderAndIsr.leader());
+        scala.collection.immutable.Set<Integer> isr = leaderAndIsr.isr().toSet();
+        List<PartitionReplica> partReplicas = new Vector<PartitionReplica>();
+        for (Object brokerObj : JavaConversions.asJavaCollection(part.getValue())) {
+          int broker = (Integer) brokerObj;
+          PartitionReplica
+                  r =
+                  new PartitionReplica(broker, (leaderAndIsr.leader() == broker), isr.contains(broker));
+          partReplicas.add(r);
+        }
+        p.setReplicas(partReplicas);
+        partitions.add(p);
+      } catch (NoSuchElementException e) {
+        log.warn("No leader and ISR information for partition {}", partId);
```

Comment:
It looks like the result of this is that we just omit the partition if it's missing leader and ISR info? Is this what we want to happen wrt REST API output? Or should the leader & ISR be null, but the `Partition` object is still returned?